modules/monitoring/library/Monitoring/Controller.php

```diff
@@ -31,22 +31,32 @@ protected function moduleInit()
 
     protected function handleFormatRequest($query)
     {
-        if ($this->_getParam('format') === 'sql') {
-            echo '<pre>'
-                . htmlspecialchars(wordwrap($query->dump()))
-                . '</pre>';
-            exit;
+        if ($this->getRequest()->getHeader('Accept') === 'application/json') {
+            $desiredFormat = 'json';
+        } elseif ($this->getRequest()->getHeader('Accept') === 'text/csv') {
+            $desiredFormat = 'csv';
+        } else {
+            $desiredFormat = strtolower($this->_getParam('format'));
         }
-        if ($this->_getParam('format') === 'json'
-            || $this->_request->getHeader('Accept') === 'application/json') {
-            header('Content-type: application/json');
-            echo json_encode($query->getQuery()->fetchAll());
-            exit;
+
+        if ($desiredFormat && $desiredFormat !== 'html' && ! $this->getRequest()->has('limit')) {
```

Comment:
Use $this->params instead and use 'html' as default when fetching the format parameter above